### PR TITLE
Move tests into simulation

### PR DESCRIPTION
**Pull request by adamchalmers**
Move test cases out of the integration tests in `tests/executor`, instead, move them into the simulation test framework under `kcl/src/simulation_tests.rs`.

### CODE REVIEW

**Review by jtran — APPROVED** [review]

File: src/wasm-lib/tests/executor/no_visuals.rs
Comment (jtran) [inline src/wasm-lib/tests/executor/no_visuals.rs:1]: Should we move these comments so we can kill this file?

Comment (adamchalmers) [reply src/wasm-lib/tests/executor/no_visuals.rs:1]: I'm just going to remove them. If you want tests that we can't currently get to pass, they should be filed as issues, not kept in comments. E.g. here's mine https://github.com/KittyCAD/modeling-app/issues/4015

Comment (jtran) [reply src/wasm-lib/tests/executor/no_visuals.rs:1]: Yes, I made an issue for it: #4432.

**Comment by adamchalmers** [comment]
Hmm. When serializing doubles, there's some unstable floating point math resulting in slight mismatches.

**Pull request merged**